### PR TITLE
Add compat data for ServiceWorkerGlobalScope

### PR DESCRIPTION
**Pull request by bunnybooboo**
https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope

as in #1002 notes for ESR might need editing.

### CODE REVIEW

**Comment by teoli2003** [comment]
This one will be a little bit more complicated as it implements mixins (GlobalFetch). I need to dig a bit to explain this completely. Stay tuned!

File: api/ServiceWorkerGlobalScope.json [hunk 1]
```diff
@@ -0,0 +1,690 @@
+{
+  "api": {
+    "ServiceWorkerGlobalScope": {
+      "__compat": {
+        "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope",
+        "support": {
+          "webview_android": {
+            "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: This should be `"40"`. 

FYI. I'm the one who originally added this information to the service worker. I was given incorrect information. Generally, the only service worker related feature that is not in Android Webview is push notifications. Please check throughout that `webview_android` has the same version number as `chrome`.

File: api/ServiceWorkerGlobalScope.json [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+            "version_added": "44"
+          },
+          "ie": {
+            "version_added": false
+          },
+          "opera": {
+            "version_added": "24"
+          },
+          "opera_android": {
+            "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 3]
```diff
@@ -0,0 +1,690 @@
...
+          "standard_track": true,
+          "deprecated": false
+        }
+      },
+      "clients": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/clients",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 4]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 5]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "registration": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/registration",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 6]
```diff
@@ -0,0 +1,690 @@
...
+            },
+            "firefox": {
+              "version_added": "44",
+              "notes": "Service workers (and Push) have been disabled in the Firefox 45 and 52 Extended Support Releases (ESR)."
+            },
+            "firefox_android": {
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 7]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "caches": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/caches",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 8]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 9]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "onactivate": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/onactivate",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 10]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 11]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "onfetch": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/onfetch",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 12]
```diff
@@ -0,0 +1,690 @@
...
+            },
+            "firefox": {
+              "version_added": "44",
+              "notes": "Service workers (and Push) have been disabled in the Firefox 45 and 52 Extended Support Releases (ESR)."
+            },
+            "firefox_android": {
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 13]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "oninstall": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/oninstall",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 14]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 15]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "onmessage": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/onmessage",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 16]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 17]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "onnotificationclick": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/onnotificationclick",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 18]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "42"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 19]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "onnotificationclose": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/onnotificationclose",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 20]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 21]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "onpush": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/onpush",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 22]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 23]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "onpushsubscriptionchange": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/onpushsubscriptionchange",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 24]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 25]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "onsync": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/onsync",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 49

File: api/ServiceWorkerGlobalScope.json [hunk 26]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

File: api/ServiceWorkerGlobalScope.json [hunk 27]
```diff
@@ -0,0 +1,690 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "skipWaiting": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/skipWaiting",
+          "support": {
+            "webview_android": {
+              "version_added": false
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 40

File: api/ServiceWorkerGlobalScope.json [hunk 28]
```diff
@@ -0,0 +1,690 @@
...
+              "version_added": "44"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerGlobalScope.json]: 24

**Pull request merged**